### PR TITLE
feat: install shell competions on HEAD

### PR DESCRIPTION
**Pull request by hanoii**
Follow up from

https://github.com/ddev/ddev/pull/5438#issuecomment-1777834147

### CODE REVIEW

**Comment by hanoii** [comment]
@rfay I am a bit confused into what bump-hombrew.sh does and how that it affects goreleaser.

I touched the fish install only to match what's on goreleaser and I wonder if _ddev for zsh should be changed either there on on go releaser (the underscore)? Not sure why it's not the same or if it affects anything.

I did try  the `make completions` (which makes ddev too so no need to run it twice) on my local formula and all worked nice.

File: .ci-scripts/bump_homebrew.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ class Ddev < Formula
     end
     bash_completion.install ".gotmp/bin/ddev_bash_completion.sh" => "ddev"
     zsh_completion.install ".gotmp/bin/ddev_zsh_completion.sh" => "_ddev"
-    fish_completion.install ".gotmp/bin/ddev_fish_completion.sh" => "ddev"
+    fish_completion.install ".gotmp/bin/ddev_fish_completion.sh" => "ddev.fisn"
```
Comment (rfay) [inline .ci-scripts/bump_homebrew.sh:72]: This is OK as this script is obsolete anyway.

**Comment by rfay** [comment]
> I am a bit confused into what bump-hombrew.sh does and how that it affects goreleaser.

I'm pretty sure that script is obsolete. Shouldn't affect goreleaser, it's pre-goreleaser.

> I wonder if _ddev for zsh should be changed either there on on go releaser (the underscore)?

Wasn't able to sort out what you were talking about here, but I seem to remember a PR that wanted _zsh for some reason.

> I did try the make completions (which makes ddev too so no need to run it twice) on my local formula and all worked nice.

I didn't realize that `completions` had `build` as a dependency, but in my comments I suggested just doing it explicitly. I guess it should be the same.

**Review by rfay — CHANGES_REQUESTED** [review]
Minor suggestions to make the build more explicit.

**Comment by hanoii** [comment]
Committed your suggestions!

> Wasn't able to sort out what you were talking about here, but I seem to remember a PR that wanted _zsh for some reason

I was referring to this (in some places you use underscore in others you don't):
- underscore
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.goreleaser.yml#L258
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.goreleaser.yml#L310
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.goreleaser.yml#L348
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.ci-scripts/bump_homebrew.sh#L71
- no underscore
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.goreleaser.yml#L196
  - https://github.com/ddev/ddev/blob/c6a8acd6762d68626fef95e408413865f76545b5/.goreleaser.yml#L231

this PR adds more lines without underscore like the ones you have above.

**Comment by rfay** [comment]
A little bit of study let me to 
* https://github.com/ddev/ddev/pull/4133

However, I don't understand the story. However, on Debian/Ubuntu, I definitely see that all the zsh completions are done with an underscore:

```
rfay@ub-2204:/usr/share/zsh/vendor-completions$ ls
_bootctl              _localectl                 _systemd-analyze
_busctl               _loginctl                  _systemd-delta
_bwrap                _networkctl                _systemd-inhibit
_curl                 _pulseaudio                _systemd-path
_ddev                 _resolvectl                _systemd-run
_docker               _sd_hosts_or_user_at_host  _systemd-tmpfiles
_dpkg-parsechangelog  _sd_outputmodes            _timedatectl
_hostnamectl          _sd_unit_files             _ubuntu-report
_j                    _snap                      _udevadm
_journalctl           _systemctl
_kernel-install       _systemd
```

**Comment by rfay** [comment]
It looks to me like perhaps the `ddev` ones are an artifact of how goreleaser deals with zsh completions, because I see the same thing in homebrew:

```
rfay@rfay-tag1-m1:/opt/homebrew/share/zsh-completions$ ls
_afew			_git-pulls		_play
_android		_git-revise		_pm2
_archlinux-java		_git-wtf		_port
_artisan		_glances		_protoc
_atach			_golang			_pygmentize
_avdmanager		_google			_qmk
_bitcoin-cli		_gpgconf		_rails
_bower			_grpcurl		_ralio
_bundle			_gtk-launch		_redis-cli
_cap			_hello			_rfkill
_cask			_hledger		_rkt
_ccache			_homestead		_rmlint
_cf			_httpie			_rslsync
_choc			_ibus			_rspec
_chromium		_include-what-you-use	_rsvm
_clang-check		_inxi			_rubocop
_clang-format		_jmeter			_sbt
_clang-tidy		_jmeter-plugins		_scala
_cmake			_jonas			_screencapture
_coffee			_jrnl			_scrub
_conan			_kak			_sdd
_concourse		_kitchen		_sdkmanager
_console		_knife			_setcap
_cppcheck		_language_codes		_setup.py
_dad			_lilypond		_sfdx
_dart			_lunchy			_shellcheck
_ddev			_mc			_showoff
_dget			_middleman		_srm
_dhcpcd			_mina			_stack
_diana			_mix			_subliminal
_direnv			_mkcert			_supervisorctl
_docpad			_mssh			_svm
_drush			_mussh			_teamocil
_ecdsautil		_mvn			_thor
_emacs			_nano			_tmuxinator
_emacsclient		_nanoc			_tmuxp
_emulator		_neofetch		_tox
_envdir			_networkQuality		_ts-node
_exportfs		_nftables		_tsc
_fab			_node			_udisksctl
_fail2ban-client	_nvm			_ufw
_ffind			_openssl		_virtualbox
_fleetctl		_openvpn3		_vnstat
_flutter		_optirun		_wemux
_fvm			_patool			_wg-quick
_fwupdmgr		_periscope		_xsel
_gas			_pgsql_utils		_yarn
_ghc			_phing			_zcash-cli
_gist			_pixz
_git-flow		_pkcon
```

**Comment by rfay** [comment]
Thanks for this, going with it. We won't know the result until next pre-release or something, but we'll fix then if needed.

**Pull request merged**